### PR TITLE
Handle auto generated legend

### PR DESCRIPTION
**Pull request by jrmi**
When we don't have label but boundaries.

### CODE REVIEW

File: src/modules/Map/InteractiveMap/components/Legend/Legend.js [hunk 1]
```diff
@@ -3,13 +3,34 @@ import PropTypes from 'prop-types';
...
 
+const computeLabel = (translate, label, boundaries, index, itemCount, rounding = null) => {
+  if (label && label !== '') {
+    return label;
+  }
+  let { lower: { value: lower }, upper: { value: upper } = {} } = boundaries;
+  if (typeof lower === 'number') {
+    if (typeof upper === 'number') { // We have two value
+      if (rounding !== null) {
+        lower = parseFloat(lower.toFixed(rounding)).toLocaleString();
```
Comment (dtrucs) [inline src/modules/Map/InteractiveMap/components/Legend/Legend.js]: `toLocaleString`?
Is it a date manipulation?
If not, `toString()` method should be more appropriate.

Comment (dtrucs) [reply src/modules/Map/InteractiveMap/components/Legend/Legend.js]: I'm not sure of that piece of code.
toFixed method should render a string value; you parseFloat it and then you restringify the whole thing.
Just `lower.toFixed(rounding)` isn't enough?

File: src/modules/Map/InteractiveMap/components/Legend/Legend.js [hunk 2]
```diff
@@ -18,7 +39,9 @@ export const Legend = ({
   position,
   content,
   history,
+  rounding,
   stackedCircles,
+  translate,
 }) => {
   const biggestDiameter = items.reduce((int, item) =>
     Math.max(int, item.diameter || DEFAULT_DIAMETER), 0);
```
Comment (dtrucs) [inline src/modules/Map/InteractiveMap/components/Legend/Legend.js]: If item.diameter equals to `0`, is it ok to have `DEFAULT_DIAMETER` value as fallback?

Comment (dtrucs) [reply src/modules/Map/InteractiveMap/components/Legend/Legend.js]: Maybe
`const biggestDiameter = items.reduce((int, { diameter = DEFAULT_DIAMETER }) => Math.max(int, diameter), 0);`

File: src/modules/Map/InteractiveMap/components/Legend/Legend.js [hunk 3]
```diff
@@ -3,13 +3,35 @@ import PropTypes from 'prop-types';
...
+import translateMock from '../../../../../utils/translate';
 import Template from '../../../../Template/Template';
 import CustomComponents from './CustomComponents';
 import Circle from './components/Circle';
 import Rect from './components/Rect';
 
 const DEFAULT_DIAMETER = 16;
 
+const computeLabel = (translate, label, boundaries, index, itemCount, rounding = null) => {
+  if (label && label !== '') {
```
[the remark this answers is not in the document]
Comment (jrmi) [reply src/modules/Map/InteractiveMap/components/Legend/Legend.js]: we don't want this test to be true if label is undefined.

File: src/modules/Map/InteractiveMap/components/Legend/Legend.js (hunk 1 above)
Comment (jrmi) [reply src/modules/Map/InteractiveMap/components/Legend/Legend.js]: For first comment, not working only with date:
image

For second comment:

File: src/modules/Map/InteractiveMap/components/Legend/Legend.js (hunk 2 above)
Comment (jrmi) [reply src/modules/Map/InteractiveMap/components/Legend/Legend.js]: Done

File: src/modules/Map/InteractiveMap/components/Legend/Legend.js (hunk 1 above)
Comment (dtrucs) [reply src/modules/Map/InteractiveMap/components/Legend/Legend.js]: My bad.
I knew https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Date/toLocaleString and not https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString.

👍

File: src/modules/Map/InteractiveMap/components/Legend/Legend.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (dtrucs) [reply src/modules/Map/InteractiveMap/components/Legend/Legend.js]: But if the value equals to an empty string (= falsy), the code will return before the second test.

why just not if(!label)?

[the remark this answers is not in the document]
Comment (dtrucs) [reply src/modules/Map/InteractiveMap/components/Legend/Legend.js]: oulala, forget it

**Pull request merged**